### DISCUSSION TITLE
Discussion on split_experiments and unindexed reflections

### DISCUSSION THREAD

**Discussion by jbeilstenedmands** [dials/dials#907] [General]
The experiment identifiers work has brought to light some potential issues when splitting experiments regarding unindexed reflections.
In a single dataset file, we have unindexed reflections denoted by a numerical id of -1, and indexed reflections by 0, 1, 2, (for each lattice). This is good for retaining the unindexed reflections (they may be able to be used later), but if one uses split_experiments, the data is split on id and the unindexed reflections have nowhere obvious to go.
Should they stay with the first dataset just by convention?
Should they be in a separate dataset, split_unindexed.refl perhaps, with a matching crystalless split_unindexed.expt (although a crystalless experiment would probably cause issues if used)?

Thoughts welcome on what would be best going forward.

**Comment by graeme-winter** [comment]
Use case:
- find spots on > 1 sweep
- index spots using multi-sweep-index
- split
- want to index unindexed spots across > 1 sweep

**Comment by graeme-winter** [comment]
Use case:
- find spots on > 1 sweep
- index as before
- split
- see indexed/unindexed reflections in RLV for sweep 1...N
- see indexed/unindexed reflections in image viewer for sweep 1...N

**Comment by graeme-winter** [comment]
Giving this further thought. I think that we actually need to change the behaviour in indexing to only change the experiment id for those reflections that are indexed, and carry the original crystal free experiment models forward. In the examples above this would mean that N sweeps go in and 2N come out, but when you split the experiments subsequently it will all make sense. In practice this is not a great deviation from the current behaviour as all we are doing is keeping around the models which belong with the unindexed spots.

[a comment by an automated account is not shown]
